Readability.js

```diff
@@ -329,10 +329,25 @@ Readability.prototype = {
   },
 
   _setNodeTag: function (node, tag) {
-    // FIXME this doesn't work on anything but JSDOMParser (ie the node's tag
-    // won't actually be set).
-    node.localName = tag.toLowerCase();
-    node.tagName = tag.toUpperCase();
+    this.log("_setNodeTag", node, tag);
+    if (node.__JSDOMParser__) {
+      node.localName = tag.toLowerCase();
+      node.tagName = tag.toUpperCase();
+      return node;
+    }
```

Comment:
I expect so, but I've not measured (which is generally true for this patch, and should probably happen before we decide to land it). I will be away from tomorrow until the second half of Monday, so if you have time to look at how this performs that would be helpful. I'm not sure what the best way of measuring this is now that both tests run in the same test framework...